### PR TITLE
TMI2-602 download all submission exports GrantExportDTO

### PR DESCRIPTION
**Pull request by rachelswart**
## Description
Adds the grantExportDTO

Ticket # and link
TMI2-631: 
https://technologyprogramme.atlassian.net/browse/TMI2-631

Summary of the changes and the related issue. List any dependencies that are required for this change:

## Type of change

Please check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [x] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End-to-End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

# Checklist:

- [ ] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [ ] I have performed a self-review of my code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have run cypress tests, and they all pass locally.

### CODE REVIEW

File: src/test/java/gov/cabinetoffice/gap/adminbackend/services/GrantExportServiceTest.java [hunk 1]
```diff
@@ -72,19 +75,34 @@ void successfullyGetGrantExports() {
                     .id(id)
                     .applicationId(1)
                     .status(GrantExportStatus.COMPLETE)
+                    .created(date)
                     .createdBy(1)
-                    .lastUpdated(Instant.now())
```
Comment (jimpurvisTCO) [inline src/test/java/gov/cabinetoffice/gap/adminbackend/services/GrantExportServiceTest.java:76]: Is there a reason the date creation was moved out from here? No big deal either way, just curious if it's a unit test quirk

**Review by jimpurvisTCO — APPROVED** [review]

File: src/main/java/gov/cabinetoffice/gap/adminbackend/services/GrantExportService.java [hunk 2]
```diff
@@ -16,12 +18,27 @@ public class GrantExportService {
...
 
-    public List<GrantExportEntity> getGrantExportsByIdAndStatus(UUID exportId, GrantExportStatus status) {
-        return exportRepository.findById_ExportBatchIdAndStatus(exportId, status);
+    public GrantExportListDTO getGrantExportsByIdAndStatus(UUID exportId, GrantExportStatus status) {
+        final List<GrantExportDTO> grantExportEntityList = exportRepository.findById_ExportBatchIdAndStatus(exportId, status)
+                .stream().map(this::mapGrantExportEntityToGrantExportDto).toList();
+        return GrantExportListDTO.builder().exportBatchId(exportId).grantExports(grantExportEntityList).build();
+    }
+
+    private GrantExportDTO mapGrantExportEntityToGrantExportDto(GrantExportEntity grantExportEntity) {
```
Comment (a-lor-cab) [inline src/main/java/gov/cabinetoffice/gap/adminbackend/services/GrantExportService.java:30]: this function could be replaced by a mapstruct mapper

File: src/test/java/gov/cabinetoffice/gap/adminbackend/services/GrantExportServiceTest.java (hunk 1 above)
Comment (a-lor-cab) [reply src/test/java/gov/cabinetoffice/gap/adminbackend/services/GrantExportServiceTest.java:76]: i think is just for reusability, as she's using the Instant.now also in the created property

Comment (rachelswart) [reply src/test/java/gov/cabinetoffice/gap/adminbackend/services/GrantExportServiceTest.java:76]: The seconds were slightly off with each Instant.now() which was causing the tests to fail

**Review by a-lor-cab — APPROVED** [review]

**Pull request merged**